File: backend/app/api/sentiment.py
from fastapi import APIRouter, HTTPException

from app.schemas.sentiment import (AnalyzeRequest, SentimentResult)
from app.services.sentiment import analyze_sentiment


router = APIRouter()

@router.post("/analyze", response_model=SentimentResult)
async def analyze(body: AnalyzeRequest) -> SentimentResult:
    try:
        return await analyze_sentiment(body.text)

    except Exception as exc:
        raise HTTPException(
            status_code=500,
            detail=f"Error interno: {exc}"
        ) from exc

File: frontend/src/services/sentiment.ts
import type { AnalyzeRequest, SentimentResult } from "../types/sentiment"

const API_URL = import.meta.env.VITE_API_URL ?? "http://localhost:8000"


export async function analyzeSentiment(text: string): Promise<SentimentResult> {
    const payload: AnalyzeRequest = {text}

    const response = await fetch(`${API_URL}/api/v1/analyze`,
        {
            method: "POST",
            headers: {
                "Content-Type": "application/json",
            },
            body: JSON.stringify(payload)
        },
    )

    if (!response.ok) {
        const error = await response.json().catch(() => null)

        throw new Error(error?.detail ?? `Request failed (${response.status})`)
    }

    return response.json()
}